File: daily_python/3105_Longest_Strictly_Increasing_or_Strictly_Decreasing_Subarray.py
import os
import sys
from typing import List

class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        ans = 1
        increase = 1
        decrease = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                increase += 1
                decrease = 1
            elif nums[i] < nums[i-1]:
                increase = 1
                decrease += 1
            else:
                increase = 1
                decrease = 1
            ans = max(ans, increase, decrease)
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.longestMonotonicSubarray(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/3105_Longest_Strictly_Increasing_or_Strictly_Decreasing_Subarray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int n = nums.size();
        int ans = 1;
        int curr = 1;
        for (int i = 1; i < n; i++){
            if (nums[i] > nums[i-1]) curr++;
            else curr = 1;
            ans = max(ans, curr);
        }
        curr = 1;
        for (int i = 1; i < n; i++){
            if (nums[i] < nums[i-1]) curr++;
            else curr = 1;
            ans = max(ans, curr);
        }
        return ans;
    }
};

int main(){
    vector<int> nums = {1,4,3,3,2};
    // vector<int> nums = {3,3,3,3};
    // vector<int> nums = {3,2,1};
    Solution S;
    cout << S.longestMonotonicSubarray(nums) << endl;
    return 0;
}